pdata/plog/log_record_flags.go

```diff
@@ -0,0 +1,40 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package plog
+
+import (
+	"go.opentelemetry.io/collector/pdata/pcommon"
+)
+
+const traceFlagsMask = uint32(0xFF)
+
+var DefaultLogRecordFlags = LogRecordFlags(0)
+
+// LogRecordFlags defines flags for the LogRecord. 8 least significant bits are the trace flags as
+// defined in W3C Trace Context specification. 24 most significant bits are reserved and must be set to 0.
+type LogRecordFlags uint32
+
+// TraceFlags copies all properties from the current struct to the dest.
+func (ms LogRecordFlags) TraceFlags() pcommon.TraceFlags {
+	return pcommon.TraceFlags(uint32(ms) & traceFlagsMask)
+}
+
+// WithTraceFlags moves all properties from the current struct to dest resetting the current instance to its zero value
```

Comment:
The comment seems incorrect. This does not reset current instance, it creates a new instance.

Reply:
Done.